### ISSUE TITLE
LTTng Produces No Events On Official Builds

### ISSUE THREAD

**Issue by brianrob** [dotnet/coreclr#27024]
On the .NET Core 3.0 official build as well as the latest official builds from master, LTTng produces no events.

I've tracked this down to the FireEtXplat* functions being emitted as (essentially) nops in the official build.

Official .NET Core 3.0 version of FireEtXplatR2RGetEntryPoint:

```
(lldb) di -n FireEtXplatR2RGetEntryPoint
libcoreclr.so`::FireEtXplatR2RGetEntryPoint(const unsigned long, PCWSTR, PCWSTR, PCWSTR, const unsigned long, const unsigned short):
    0x7ffff7382ec0 <+0>: xorl   %eax, %eax
    0x7ffff7382ec2 <+2>: retq
```

Private build of .NET Core 3.0 (same commit hash) of FireEtXplatR2RGetEntryPoint:

```
(lldb) di -n FireEtXplatR2RGetEntryPoint
libcoreclr.so`::FireEtXplatR2RGetEntryPoint(const unsigned long, PCWSTR, PCWSTR, PCWSTR, const unsigned long, const unsigned short):
    0x7ffff7366e80 <+0>:    pushq  %rbp
    0x7ffff7366e81 <+1>:    movq   %rsp, %rbp
    0x7ffff7366e84 <+4>:    pushq  %r15
    0x7ffff7366e86 <+6>:    pushq  %r14
    0x7ffff7366e88 <+8>:    pushq  %r13
    0x7ffff7366e8a <+10>:   pushq  %r12
    0x7ffff7366e8c <+12>:   pushq  %rbx
    0x7ffff7366e8d <+13>:   subq   $0x3b8, %rsp              ; imm = 0x3B8
    0x7ffff7366e94 <+20>:   movq   %rcx, -0x3b8(%rbp)
    0x7ffff7366e9b <+27>:   movq   %rdx, -0x3c0(%rbp)
    0x7ffff7366ea2 <+34>:   movq   %rdi, -0x3d0(%rbp)
    0x7ffff7366ea9 <+41>:   movq   %fs:0x28, %rax
    0x7ffff7366eb2 <+50>:   movq   %rax, -0x30(%rbp)
    0x7ffff7366eb6 <+54>:   cmpl   $0x0, 0x25ebab(%rip)      ; __tracepoint_DotNETRuntime___R2RGetEntryPoint + 7
    0x7ffff7366ebd <+61>:   je     0x7ffff7366f1c            ; <+156> at stackstring.hpp
    0x7ffff7366ebf <+63>:   movl   %r9d, %r14d
    0x7ffff7366ec2 <+66>:   movq   %r8, %r13
    0x7ffff7366ec5 <+69>:   leaq   -0x150(%rbp), %r12
    0x7ffff7366ecc <+76>:   movq   %r12, -0x48(%rbp)
    0x7ffff7366ed0 <+80>:   movl   $0x105, %eax              ; imm = 0x105
    0x7ffff7366ed5 <+85>:   movq   %rax, %xmm0
    0x7ffff7366eda <+90>:   movdqu %xmm0, -0x40(%rbp)
    0x7ffff7366edf <+95>:   movq   %rsi, -0x3c8(%rbp)
    0x7ffff7366ee6 <+102>:  movq   %rsi, %rdi
    0x7ffff7366ee9 <+105>:  callq  0x7ffff72e4da0            ; ::PAL_wcslen(const wchar_16 *) at wchar.cpp:664
    0x7ffff7366eee <+110>:  leal   (%rax,%rax), %eax
    0x7ffff7366ef1 <+113>:  addl   $0x2, %eax
    0x7ffff7366ef4 <+116>:  movslq %eax, %r15
    0x7ffff7366ef7 <+119>:  movq   -0x48(%rbp), %rbx
    0x7ffff7366efb <+123>:  testq  %rbx, %rbx
    0x7ffff7366efe <+126>:  je     0x7ffff7366f36            ; <+182> [inlined] StackString<260ul, char>::Resize(unsigned long) at stackstring.hpp:159
    0x7ffff7366f00 <+128>:  cmpq   %r15, -0x40(%rbp)
    0x7ffff7366f04 <+132>:  movl   %r14d, -0x3a8(%rbp)
    0x7ffff7366f0b <+139>:  movl   %eax, -0x3b0(%rbp)
    0x7ffff7366f11 <+145>:  jbe    0x7ffff7366f50            ; <+208> [inlined] StackString<260ul, char>::Resize(unsigned long) + 26 at stackstring.hpp:159
    0x7ffff7366f13 <+147>:  movq   %r15, -0x38(%rbp)
…snip…
```

**Comment by brianrob** [comment]
@noahfalk, @sywhang, @josalem

**Comment by sywhang** [comment]
Reopening for 3.0/3.1 port

**Comment by tommcdon** [comment]
Issue is fixed in 3.1.  Closing 3.0 backporting tracking bug as we do not currently have customer requests to backport
